### PR TITLE
chore(tokens): updates font related tokens and adds typography semantic tokens

### PR DESCRIPTION
**Pull request by pixelflips**
# Description

To future-proof the token values, we will increase the scale of the current font sizes and line height values to provide more flexibility. Some values may not be used immediately but will be available if and when necessary.

These new values will also be used to set up the typography semantic tokens.

CORE:
Updates font weight
Updates line height
Updates font sizes
Fixes letter spacing

Semantic:
Add typography tokens based on core updates above

Fixes [DSS-586](https://kajabi.atlassian.net/browse/DSS-586)

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [x] other: Token Studio in Figma

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-586]: https://kajabi.atlassian.net/browse/DSS-586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/core/src/global/styles/tokens/tokens.json [hunk 1]
```diff
@@ -751,6 +819,154 @@
...
       }
+    },
+    "heading-1": {
+      "value": {
+        "fontWeight": "{font-weight.bold}",
+        "fontSize": "{font-size.250}",
+        "lineHeight": "{line-height.275}",
+        "letterSpacing": "{letter-spacing.md}"
+      },
+      "type": "typography"
```
Comment (anechol) [inline libs/core/src/global/styles/tokens/tokens.json]: We should be able to add the font family in here as well in TS.

**Review by anechol — APPROVED** [review]
One small thing to change, but otherwise, awesome work getting these updated.

File: libs/core/src/global/styles/tokens/tokens.json (hunk 1 above)
Comment (pixelflips) [reply libs/core/src/global/styles/tokens/tokens.json]: @anechol all should be set to `{font-family.circular}` now. Great catch!

**Review by QuintonJason — APPROVED** [review]
Good to go

**Comment by ju-Skinner** [comment]
Shouldn't the `core.scss` and `semantic.scss` files be updated in the PR as well?

**Comment by pixelflips** [comment]
> Shouldn't the `core.scss` and `semantic.scss` files be updated in the PR as well?

My understanding is that happens when the build tokens is run, which right now we are doing locally/manually. @anechol can confirm as I've only been updating from within token studio right now.

**Comment by ju-Skinner** [comment]
Last Friday we talked about this. I mentioned how the build would potentially have conflicts due to files changing and it would put us in a weird scenario. I proposed that we should make those changes and have them be a part of the PR (this). Again we can reconvene as a team and discuss further. 

cc: @monicawheeler @QuintonJason @anechol

**Comment by pixelflips** [comment]
Ok cool, got it. I can run here for now so we can get these in.

**Review by anechol — APPROVED** [review]
Nice work!

**Review by ju-Skinner — APPROVED** [review]

File: libs/core/src/global/styles/tokens/core/_core.scss [hunk 2]
```diff
@@ -1,13 +1,21 @@
 /**
  * Do not edit directly
- * Generated on Fri, 12 Apr 2024 15:37:39 GMT
+ * Generated on Wed, 24 Apr 2024 16:03:41 GMT
  */
```
Comment (ju-Skinner) [inline libs/core/src/global/styles/tokens/core/_core.scss:4]: Note for @anechol. We need to figure out how to prevent this from being added.

Comment (anechol) [reply libs/core/src/global/styles/tokens/core/_core.scss:4]: Yes this is something I will look into.

**Pull request merged**